src/game/AI/ScriptDevAI/scripts/kalimdor/darkshore.cpp

```diff
@@ -831,28 +840,49 @@ struct npc_corrupted_furbolgAI : public ScriptedAI
                 m_creature->CombatStop();
 
                 float x, y, z, angle;
-                angle = m_creature->GetAngle(bowlCoords.GetPositionX(), bowlCoords.GetPositionY());
-                m_creature->GetNearPointAt(bowlCoords.GetPositionX(), bowlCoords.GetPositionY(), bowlCoords.GetPositionZ(),
+                angle = m_creature->GetAngle(m_bowlCoords.GetPositionX(), m_bowlCoords.GetPositionY());
+                m_creature->GetNearPointAt(m_bowlCoords.GetPositionX(), m_bowlCoords.GetPositionY(), m_bowlCoords.GetPositionZ(),
                     m_creature, x, y, z, m_creature->GetObjectBoundingRadius(), CONTACT_DISTANCE * 2.f, angle + M_PI);
                 m_creature->SetWalk(false);
                 m_creature->GetMotionMaster()->MovePoint(1, x, y, z);
             }
         });
-        Reset();
+        AddTimerlessCombatAction(ACTION_FLEE, true);
+        AddCombatAction(ACTION_THUNDERCLAP, true);
+        AddCombatAction(ACTION_HEALING_WARD, true);
     }
 
     void Reset() override
     {
+        CombatAI::Reset();
+        
+        switch (m_creature->GetEntry())
+        {
+            case NPC_BLACKWOOD_WARRIOR: ResetIfNotStarted(ACTION_THUNDERCLAP, GetSubsequentActionTimer(ACTION_THUNDERCLAP)); DoCastSpellIfCan(m_creature, SPELL_BATTLE_STANCE); break;
```

Comment:
Instead of doing this, and doing AddCombatAction(ACTION_THUNDERCLAP, true);

do AddCombatAction(ACTION_THUNDERCLAP, initialMin, initialMax); and each reset CombatAI::Reset will do it for you :) I can change it when merging but making sure you know.

Reply:
Wouldn't both types of furbolg have the same spellset?
Only totemics are supposed to use healing wards and only warriors are supposed to use thunderclap

Reply:
Well if you add a condition to only call add for one or the other then they wont have both.

Reply:
Well if you wanna be extra modern, add the spells to creature_spell_list :D And you dont have this problem to begin with.